lte/gateway/python/magma/pipelined/app/restart_mixin.py

```diff
@@ -126,7 +129,13 @@ def handle_restart(self, requests) -> SetupFlowsResult:
                 tbl,
                 [flow.match for flow in startup_flows_map[tbl]],
             )
-        self._remove_extra_flows(startup_flows_map)
+        service = MagmaService('pipelined', mconfigs_pb2.PipelineD())
+        service_config = service.config
+
+        if ((self.APP_NAME == "enforcement") and (service_config['redis_enabled'] == True)):
```

Comment:
@aniket021997 you requested a re-review - this point is still open. Do you have an opinion?

Reply:
Hi @aniket021997 , technically this works, but imho this still causes a forced connection between orthogonal classes in the inheritance structure.
What I would suggest:

Create a function in restart_mixin.py
```
  skip_extra_flows_removal(self):
    ''' some documentation that this can be overridden by controllers and what this is causing '''
    return False
```
    
enforcement.py overrides this function:
```
  skip_extra_flows_removal(self):
    ''' some documentation why extra flows should not be deleted here if staless is True '''
    return self.is_stateless
```

in restart_mixin.py use the function in handle_restart
```
  if self.skip_extra_flows_removal():
    logging ...
  else
    remove extra flows
```

Also by this you should not need to touch other controllers.

Reply:
Thank you for changing this.
Does it make sense to have a test for this feature? I think a reduced version of `lte/gateway/python/magma/pipelined/tests/test_restart_resilience.py` `test_enforcement_restart` with `'redis_enabled': True` can be used here.

Reply:
@nstng thanks for the review comment. We added redis_enabled flag under config in test cases since make test_python was failing without redis_enabled flag with - "Failed to start test apps in separate thread: Ryu apps launch exception: 'redis_enabled'"

Reply:
@aniket021997, yes, but my point was that a test for the feature added here could make sense, i.e., a test that ...
* sets `'redis_enabled': True,`
* adds extra flows to the enforcement controller
* asserts that all flows are present
* restarts the controllers
* assert that extra flows are still present

The test that I mentioned above could be used as a blueprint.

Reply:
@nstng thanks for the review. We have added test test_with_stateless_enforcement_restart with one imsi and one policy.